Geometry/TrackerCommonData/plugins/DDTrackerIrregularRingAlgo.cc

```diff
@@ -0,0 +1,280 @@
+  <rParent name="tracker:Ring5Layer1Plus"/>
+  <String name="ChildName" value="tracker:BModule5Layer1"/>
+  <Numeric name="N" value="9"/>
+  <Numeric name="StartCopyNo" value="1"/>
+  <Numeric name="IncrCopyNo" value="2"/>
+  <Numeric name="RangeAngle" value="360*deg"/>
+  <Numeric name="StartAngle" value="90*deg"/>
+  <Numeric name="Radius" value="247"/>
+  <Vector name="Center" type="numeric" nEntries="3">0,0,-5.45415</Vector>
+  <Numeric name="IsZPlus" value="1"/>
+  <Numeric name="TiltAngle" value="47*deg"/>
+  <Numeric name="IsFlipped" value="1"/>
+  </Algorithm>
+*/
+
+using namespace std;
+
+class DDTrackerIrregularRingAlgo : public DDAlgorithm {
+public:
+  // Constructor and Destructor
+  DDTrackerIrregularRingAlgo();
+  ~DDTrackerIrregularRingAlgo() override;
+
+  void initialize(const DDNumericArguments& nArgs,
+                  const DDVectorArguments& vArgs,
+                  const DDMapArguments& mArgs,
+                  const DDStringArguments& sArgs,
+                  const DDStringVectorArguments& vsArgs) override;
+
+  void execute(DDCompactView& cpv) override;
+
+private:
+  int n;                  //Number of copies
+  int startCopyNo;        //Start Copy number
+  int incrCopyNo;         //Increment in Copy number
+  double rangeAngle;      //Range in Phi angle
+  double startAngle;      //Start Phi angle
+  double radius;          //Radius
+  vector<double> center;  //Phi values
+  vector<double> phiAngles;
+  vector<double> radiusValues;
+  vector<double> yawAngles;
+  bool isZPlus;      //Is Z positive ?
+  double tiltAngle;  //Module's tilt angle (absolute value)
+  bool isFlipped;    //Is the module flipped ?
+  double delta;      //Increment in Phi
+
+  string idNameSpace;  //Namespace of this and ALL sub-parts
+  string childName;    //Child name
+};
+
+DDTrackerIrregularRingAlgo::DDTrackerIrregularRingAlgo() {
+  LogDebug("TrackerGeom") << "DDTrackerIrregularRingAlgo info: Creating an instance";
+}
+
+DDTrackerIrregularRingAlgo::~DDTrackerIrregularRingAlgo() {}
+
+void DDTrackerIrregularRingAlgo::initialize(const DDNumericArguments& nArgs,
+                                            const DDVectorArguments& vArgs,
+                                            const DDMapArguments&,
+                                            const DDStringArguments& sArgs,
+                                            const DDStringVectorArguments&) {
+  n = int(nArgs["N"]);
+  startCopyNo = int(nArgs["StartCopyNo"]);
+  incrCopyNo = int(nArgs["IncrCopyNo"]);
+  rangeAngle = nArgs["RangeAngle"];
+  startAngle = nArgs["StartAngle"];
+  radius = nArgs["Radius"];
+  center = vArgs["Center"];
+  yawAngles = vArgs["yawAngleValues"];
+  phiAngles = vArgs["phiAngleValues"];
+  radiusValues = vArgs["radiusValues"];
+  isZPlus = bool(nArgs["IsZPlus"]);
+  tiltAngle = nArgs["TiltAngle"];
+  isFlipped = bool(nArgs["IsFlipped"]);
+
+  if (fabs(rangeAngle - 360.0 * CLHEP::deg) < 0.001 * CLHEP::deg) {
```

Comment:
```suggestion
  if (std::abs(rangeAngle - 2._pi) < 1.7e-5) {  // 1.7e-5 radians is about 0.001 degrees
```

Reply:
I removed this code snippet as it was calculating a variable that was not actually getting used (leftover from when the algorithm was dual-purpose)